File: FamilyTools.Web/src/app/models/account-enter.ts
import { BaseModel } from './base-model';
import {AccountLine, EMPTY_LINE_FORM_MODEL, EMPTY_LIST_LINE_FORM_MODEL} from './account-line';
import { AccountTag } from './account-tag';
import { OperationType } from '../enum/app.enum';

export interface AccountEnter extends BaseModel {
  lines: AccountLine[];
  tag: AccountTag;
  name: string;
  operationType: OperationType;
  totalValue: number;
  date: Date;
  isDisabled: boolean;
}

export const EMPTY_ENTER_FORM_MODEL = {
  lines: EMPTY_LIST_LINE_FORM_MODEL,
  tag: {
    name: "",
    color: ""
  },
  name: '',
  operationType: OperationType.Unknown,
  totalValue: 0,
  date: new Date(),
  isDisabled: false
};

File: FamilyTools.Data/Models/EasyCompta/AccountEnter.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace FamilyTools.Data.Models.EasyCompta;

[Table("AccountEnters")]
public class AccountEnter : BaseModel
{
    public AccountEnter()
    {
        if (this.Lines?.Count > 0) this.TotalValue = this.Lines.Sum(line => line.Value);
    }

    public AccountEnter(string name, OperationType operationType, DateOnly date, float totalValue,
        List<AccountLine> lines, AccountTag tag)
    {
        this.Name = name;
        this.OperationType = operationType;
        this.TotalValue = totalValue;
        this.Date = date;
        this.Lines = lines;
        this.Tag = tag;
    }

    public ICollection<AccountLine> Lines { get; set; } = new List<AccountLine>();
    public AccountTag Tag { get; set; } = new();
    public OperationType OperationType { get; set; } = new();
    public string Name { get; set; } = string.Empty;
    public float TotalValue { get; set; }
    public DateOnly Date { get; set; }
    public int PageId { get; set; }
    public bool IsDisabled { get; set; }

    [JsonIgnore] public AccountPage Page { get; set; } = null!;
}